### PR TITLE
support for gitlab

### PR DESCRIPTION
**Pull request by wanghao75**

### CODE REVIEW

File: robot-gitlab-framework/dispatcher.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+	return eventType
+}
+
+func (d *dispatcher) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	eventType, eventGUID, payload, ok := parseRequest(w, r)
+	if !ok {
+		return
+	}
+
+	eventType = convertEventType(eventType, payload)
```
Comment (zengchen1024) [inline robot-gitlab-framework/dispatcher.go:65]: why shoud invoke this function

File: options/gitlab.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
+package options
+
+import (
+	"flag"
+)
+
+// GitLabOptions holds options for interacting with GitLab.
+type GitLabOptions struct {
+	TokenPath     string
+	RepoCacheDir  string
```
Comment (zengchen1024) [inline options/gitlab.go:10]: no need RepoCacheDir and CacheRepoOnPV

File: gitlabclient/client.go [hunk 3]
```diff
@@ -0,0 +1,902 @@
+package gitlabclient
+
+import (
+	"fmt"
+	"github.com/xanzy/go-gitlab"
+	"strings"
```
Comment (zengchen1024) [inline gitlabclient/client.go:6]: "fmt"
	"strings"

	"github.com/xanzy/go-gitlab"

**Review by zengchen1024 — CHANGES_REQUESTED** [review]

File: robot-gitlab-framework/dispatcher.go (hunk 1 above)
Comment (wanghao75) [reply robot-gitlab-framework/dispatcher.go:65]: 区分评论事件是MR还是Issue

**Pull request merged**